cmd/crane/cmd/root.go

```diff
@@ -146,3 +152,27 @@ func (ht *headerTransport) RoundTrip(in *http.Request) (*http.Response, error) {
 	}
 	return ht.inner.RoundTrip(in)
 }
+
+type warnCollector struct {
+	warns map[string]struct{}
+}
+
+func (d *warnCollector) Add(warning string) {
+	d.warns[warning] = struct{}{}
```

Comment:
one thing to watch out for is accumulation without cleanup in a long-running process... if the lifetime of the collector is expected to be ~short, this is probably fine. if the lifetime could be long-running, this could grow without bound

Reply:
In `crane`'s context I think we can expect this to be pretty short-lived, most invocations are at most "copy this image to this image" and there's not a lot of opportunity for it to do anything else.

But I agree this is a good thing to consider for long-running things that may collect logs of warnings.